roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2

```diff
@@ -37,14 +37,15 @@
       {% set ipxe_rom_filename = (ipxe_architecture + '/' + ipxe_embed + '_' + ipxe_driver + 'ipxe.efi') %}
     {% else %}  {# Assume everything else is pcbios #}
       {% set ipxe_rom_filename = (ipxe_architecture + '/' + ipxe_embed + '_undionly.kpxe') %}
-    {% endif %}
+    {% endif -%}
 
-    {% for nic, nic_args in hostvars[host]['network_interfaces'].items() %}
+    {%- for nic, nic_args in hostvars[host]['network_interfaces'].items() %}
```

Comment:
I'm still not sure this minus sign at the beginning of the block changes anything in the output.
Can you confirm the output file is different without this change?

Reply:
Dear @actatux,
I will trying without this minus sign.
I'll keep you informed.
Regards

Reply:
Hello @actatux,
The minus sign adding causes change.
Regards

Reply:
Thank you @hmeAtos. I was curious about the change, so I ran a test on top of this PR, with and without the minus sign at lines 42 and 82.
The main changes are the order of the hosts in the files and a few trailing whitespaces moving from one line to the other.

I can reproduce the same behaviour without modifying your PR, after 2 consecutive runs.

```
$ md5sum /tmp/dhcpd.subnet.conf.?
b890fecbb81697e7a04bbca02d555bbe  dhcpd.subnet.conf.1
9311154d685103f81048a203cd640742  dhcpd.subnet.conf.2
$ grep -c '^ \+$' dhcpd.subnet.conf.?
4
```

You can remove the trailing whitespaces with either:
```diff
--- a/roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2
+++ b/roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2
@@ -20,12 +20,12 @@
       {% set ipxe_driver = 'snponly_' %}
     {% elif hostvars[host]['equipment_profile']['ipxe_driver'] == 'snp' %}
       {% set ipxe_driver = 'snp_' %}
-    {% else %}  {# Assume everything else is default driver #}
+    {% else %}{# Assume everything else is default driver #}
       {% set ipxe_driver = '' %}
     {% endif %}
     {% if hostvars[host]['equipment_profile']['hardware']['cpu']['architecture'] == 'arm64' %} {# Now select architecture #}
       {% set ipxe_architecture = 'arm64' %}
-    {% else %}  {# Assume everything else is x86_64 #}
+    {% else %}{# Assume everything else is x86_64 #}
       {% set ipxe_architecture = 'x86_64' %}
     {% endif %}
     {% if hostvars[host]['equipment_profile']['ipxe_embed'] == 'dhcpretry' %} {# Now select embed script #}
@@ -35,7 +35,7 @@
     {% endif %}
     {% if hostvars[host]['equipment_profile']['ipxe_platform'] == 'efi' %} {# Finally, build file name depending of platform and previous parameters #}
       {% set ipxe_rom_filename = (ipxe_architecture + '/' + ipxe_embed + '_' + ipxe_driver + 'ipxe.efi') %}
-    {% else %}  {# Assume everything else is pcbios #}
+    {% else %}{# Assume everything else is pcbios #}
       {% set ipxe_rom_filename = (ipxe_architecture + '/' + ipxe_embed + '_undionly.kpxe') %}
     {% endif -%}
```
or
```diff
--- a/roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2
+++ b/roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2
@@ -20,12 +20,12 @@
       {% set ipxe_driver = 'snponly_' %}
     {% elif hostvars[host]['equipment_profile']['ipxe_driver'] == 'snp' %}
       {% set ipxe_driver = 'snp_' %}
-    {% else %}  {# Assume everything else is default driver #}
+    {% else %}  {#- Assume everything else is default driver #}
       {% set ipxe_driver = '' %}
     {% endif %}
     {% if hostvars[host]['equipment_profile']['hardware']['cpu']['architecture'] == 'arm64' %} {# Now select architecture #}
       {% set ipxe_architecture = 'arm64' %}
-    {% else %}  {# Assume everything else is x86_64 #}
+    {% else %}  {#- Assume everything else is x86_64 #}
       {% set ipxe_architecture = 'x86_64' %}
     {% endif %}
     {% if hostvars[host]['equipment_profile']['ipxe_embed'] == 'dhcpretry' %} {# Now select embed script #}
@@ -35,7 +35,7 @@
     {% endif %}
     {% if hostvars[host]['equipment_profile']['ipxe_platform'] == 'efi' %} {# Finally, build file name depending of platform and previous parameters #}
       {% set ipxe_rom_filename = (ipxe_architecture + '/' + ipxe_embed + '_' + ipxe_driver + 'ipxe.efi') %}
-    {% else %}  {# Assume everything else is pcbios #}
+    {% else %}  {#- Assume everything else is pcbios #}
       {% set ipxe_rom_filename = (ipxe_architecture + '/' + ipxe_embed + '_undionly.kpxe') %}
     {% endif -%}
```

I also notice an alignment issue:
```
  host lawfer {
    option host-name "lawfer";
    hardware ethernet 08:00:27:0d:f8:a3;
    fixed-address 10.13.3.2;
    filename "x86_64/standard_undionly.kpxe";
  }

    host blawfer {
      option host-name "blawfer";
      hardware ethernet 08:00:27:0d:f8:b3;
      fixed-address 10.13.103.2;
    }
```

Reply:
Hi @hmeAtos ,
Could you find some time to apply above recommendation next week?

Reply:
Hi @actatux and @oxedions,
I have apply this recommendation here : [PR:133](https://github.com/bluebanquise/bluebanquise/pull/133)

Indeed, these trailling spaces have caused problems. Now, they are fixed.
I suggest you apply my two PRs to fix the role advanced_dhcp_server.
Regards,

Reply:
Merging one will probably need to rebase the other. Stay tuned!

Reply:
> > > Dear @hmeAtos
> > > Could you also correct roles/core/dhcp_server/templates/dhcpd.subnet.conf.j2 as suggested by @actatux ?
> > > So I can merge this :-)
> > > Ben
> > 
> > 
> > Done.
> 
> Where?

here: https://github.com/bluebanquise/bluebanquise/pull/133/commits/08170505dae4cb0471f6ddd7bb10b7ad2bce95c8